integration/e2e_test.go

```diff
@@ -398,52 +296,67 @@ func installCilium(t testing.TB) {
+		return machines > 0
+	}
+	if err := retryableCommandWithConditionFunction(t, command, testFn); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log("Checking Cilium daemon set")
+	command = []string{"kubectl", "--kubeconfig=" + KUBECONFIG, "-n", "kube-system", "wait", "ds/cilium", fmt.Sprintf("--for=jsonpath={.status.numberAvailable}=%d", machines)}
+	if err := retryableCommand(t, command); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// retryableCommand runs command and retires if command fails.
+// Runs provided command, if commands return err retries it retryMaxAttempts and waits secondsBetweenAttempts between next execution.
+// If after all attempts error occurs error gets returned.
+func retryableCommand(t testing.TB, command []string) error {
+	var err error
+	var outputBytes []byte
+	for attempt := 0; attempt < retryMaxAttempts; attempt++ {
+		if attempt > 1 {
+			t.Logf("Retrying")
+			time.Sleep(secondsBetweenAttempts * time.Second)
+		}
 		t.Logf("running command: %s", strings.Join(command, " "))
 		cmd := exec.Command(command[0], command[1:]...)
-		outputBytes, err := cmd.CombinedOutput()
-		output := string(outputBytes)
-		if err != nil {
-			t.Log(output)
-			if attempt >= maxAttempts {
-				t.Fatal(err)
-			} else {
-				attempt++
-				t.Log("Retrying")
-				time.Sleep(10 * time.Second)
-			}
-		} else {
-			machines = strings.Count(output, "\n")
-			if machines > 0 {
-				break
-			}
+		outputBytes, err = cmd.CombinedOutput()
+		t.Log(string(outputBytes))
+		if err == nil {
+			break
 		}
 	}
+	return err
+}
 
-	t.Log("Checking Cilium daemon set")
-	command = []string{"kubectl", "--kubeconfig=" + KUBECONFIG, "-n", "kube-system", "wait", "ds/cilium", fmt.Sprintf("--for=jsonpath={.status.numberAvailable}=%d", machines)}
-	for {
+// retryableCommandWithConditionFunction runs command with retry.
+// Runs provided command, if commands return err retries it retryMaxAttempts and waits secondsBetweenAttempts between next execution.
+// Additionally, to finish correctly runs function fn func(string) bool, if passed function returns true
+// the retryableCommandWithConditionFunction returns nil
+func retryableCommandWithConditionFunction(t testing.TB, command []string, fn func(string) bool) error {
+	var err error
+	var outputBytes []byte
+	for attempt := 0; attempt < retryMaxAttempts; attempt++ {
+		if attempt > 1 {
+			t.Logf("Retrying")
+			time.Sleep(secondsBetweenAttempts * time.Second)
+		}
+		t.Logf("running command: %s", strings.Join(command, " "))
 		cmd := exec.Command(command[0], command[1:]...)
-		outputBytes, err := cmd.CombinedOutput()
-		if err != nil {
-			t.Logf("running command: %s", strings.Join(command, " "))
-			t.Log(string(outputBytes))
-			if attempt >= maxAttempts {
-				t.Fatal(err)
-			} else {
-				attempt++
-				t.Log("Retrying")
-				time.Sleep(20 * time.Second)
-			}
-		} else {
+		outputBytes, err = cmd.CombinedOutput()
+		output := string(outputBytes)
+		t.Log(output)
+		if err == nil && fn(output) {
```

Comment:
Can we give fn a more descriptive name? I'd have to go look at the implementation of fn to know what it does here, which we'd like to avoid.